### ISSUE TITLE
Website button issues

### ISSUE THREAD

**Issue by baimafeima** [solus-project/solus-sc#56]
I think it is a good idea to refer to the website of a particular program in the SC. Currently, this is realized through the _Website_ button. However, I have the following issues with it:

- Clicking the button directly links to a website and automatically launches the default browser. I would prefer to first see what website it links to and decide for myself whether I click it or not and with what browser to open it. 
- It is inconsistently applied, that is, the button is sometimes displayed and sometimes not. I think the user interface should not be dependent on whether a particular website is available to be linked to or not.

Possible solutions:
- A _Details_ button instead of a _Website_ button could reveal more information than just the website link and be permanently displayed (due to being more general), thereby avoiding inconsistencies in the interface design when no website is available for a particular program. 
- Clicking a button could _slide in_ additional information. I would naturally expect this behaviour rather than having my default browser launched automatically. This would give the user an additional amount of control over what happens when clicking the button.

**Comment by ikeydoherty** [comment]
> It is inconsistently applied, that is, the button is sometimes displayed and sometimes not. I think the user interface should not be dependent on whether a particular website is available to be linked to or not.

That's because not all packages contain website information.

> Clicking the button directly links to a website and automatically launches the default browser.

I.. see no issue with this. You click a button called website you kinda expect to be taken to the website, no?

**Comment by baimafeima** [comment]
> You click a button called website you kinda expect to be taken to the website, no?

Probably. I think an implementation as part of a _Details_ button would allow to have more _optional_ information displayed such as website, Github et al., issue tracker, main developer name, Patreon page (if applicable). The particular use cases I see are: 

1. Extremely efficient to follow up on upstream issues
1. A non-obtrusive way to encourage financial support for FOSS projects

**Comment by ikeydoherty** [comment]
AppStream information provides us with (potentially):

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html

- homepage
- bugtracker
- donation

In terms of applicable URLs for the web UI. Hiding these behind an optional dialog is a disservice to the developers. Having a better UI that relied more on iconography than text would be more appropriate imo.

**Comment by baimafeima** [comment]
> Hiding these behind an optional dialog is a disservice to the developers. Having a better UI that relied more on iconography than text would be more appropriate imo.

Agreed. I think a left-click on such an icon could lead directly to the page and a right-click could offer a more specific _launch with ..._ option.
